sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaTranslation.java

```diff
@@ -429,15 +442,19 @@ private static FieldType fieldTypeFromProtoWithoutNullable(SchemaApi.FieldType p
         } else if (urn.equals(URN_BEAM_LOGICAL_DECIMAL)) {
           return FieldType.DECIMAL;
         } else if (urn.startsWith("beam:logical_type:")) {
-          try {
-            return FieldType.logicalType(
-                (LogicalType)
-                    SerializableUtils.deserializeFromByteArray(
-                        logicalType.getPayload().toByteArray(), "logicalType"));
-          } catch (IllegalArgumentException e) {
-            LOG.warn(
-                "Unable to deserialize the logical type {} from proto. Mark as UnknownLogicalType.",
-                urn);
+          if (!logicalType.getPayload().isEmpty()) {
```

Comment:
Note for myself - I see verbose log here. We should cache the UnknownLogicalType if it has a urn.

Reply:
leave as future improvement - currently change to DEBUG log to reduce noise

Reply:
Can we raise an issue and add a TODO here to make sure it isn't lost?